partiql-types/src/main/kotlin/org/partiql/types/SchemaType.kt

```diff
@@ -0,0 +1,9 @@
+package org.partiql.types
+
+/**
+ * Represents a `Schema` (Shape of data) as a bag; this is still experimental
```

Comment:
> Do we allow the underlying data to be bag of atomic value ?
Yes.

> This can not be the underlying data format.
Can you explain this further?

Let's discuss the rest further as I need more clarity on them.